src/mpi/datatype/typerep/src/typerep_yaksa_pack.c

```diff
@@ -323,3 +531,276 @@ int MPIR_Typerep_wait(MPIR_Typerep_req typerep_req)
   fn_fail:
     goto fn_exit;
 }
+
+int MPIR_Typerep_iunpack_reduce(const void *inbuf, MPI_Aint insize, void *outbuf, MPI_Aint outcount,
+                                MPI_Datatype datatype, MPI_Aint outoffset,
+                                MPI_Aint * actual_unpack_bytes, MPIR_Typerep_req * typerep_req,
+                                MPI_Op op)
+{
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIR_TYPEREP_IUNPACK_REDUCE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIR_TYPEREP_IUNPACK_REDUCE);
+
+    int mpi_errno = MPI_SUCCESS;
+    int rc;
+
+    if (outcount == 0) {
+        *actual_unpack_bytes = 0;
+        goto fn_exit;
+    }
+
+    MPIR_Assert(datatype != MPI_DATATYPE_NULL);
+
+    yaksa_type_t type = MPII_Typerep_get_yaksa_type(datatype);
+    yaksa_op_t yaksa_op = MPII_Typerep_get_yaksa_op(op);
+
+    uintptr_t size;
+    rc = yaksa_type_get_size(type, &size);
+    MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa");
+
+    uintptr_t real_insize = MPL_MIN(insize, size * outcount);
+
+    uintptr_t real_unpack_bytes;
+    rc = yaksa_iunpack(inbuf, real_insize, outbuf, outcount, type, outoffset, &real_unpack_bytes,
+                       NULL, yaksa_op, (yaksa_request_t *) typerep_req);
+    MPIR_ERR_CHKANDJUMP(rc, mpi_errno, MPI_ERR_INTERN, "**yaksa");
+
+  fn_exit:
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIR_TYPEREP_IUNPACK_REDUCE);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+int MPIR_Typerep_rma_reduce(void *source_buf, int source_count, MPI_Datatype source_dtp,
+                            void *target_buf, int target_count, MPI_Datatype target_dtp,
+                            MPI_Op acc_op, int src_kind)
```

Comment:
This is fine. I would like this function be refactored a bit to read more easily. But that can be done separately.